### PR TITLE
[CLOUDTRUST-3991] kafka audit event persistency

### PR DESCRIPTION
**Pull request by basbeu**

### CODE REVIEW

**Review by harture — APPROVED** [review]

**Review by harture — CHANGES_REQUESTED** [review]
A few details but the censor of the secrets is important

File: events/kafkaproducer.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+func GetKafkaProducerConfig(c cs.Configuration, prefix string) KafkaProducerConfig {
+	var cfg KafkaProducerConfig
+
+	cfg.Noop = !c.GetBool(prefix)
+
+	if !cfg.Noop {
+		cfg.Version = c.GetString(prefix + "-version")
+		cfg.Brokers = c.GetStringSlice(prefix + "-brokers")
+		cfg.ClientID = c.GetString(prefix + "-client-id")
+		cfg.ClientSecret = c.GetString(prefix + "-client-secret")
```
[the remark this answers is not in the document]
Comment (basbeu) [reply events/kafkaproducer.go:30]: Binding with env variable done at the component level

**Review by fperot74 — APPROVED** [review]

File: events/events.go [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+		eventType: eventType,
+		details:   details,
+	}
+}
+
+func NewEventOnUser(origin string, eventType string, agentRealmName string, agentUserID string, agentUsername string, targetRealmName string, targetUserID string, targetUserName string, details map[string]string) Event {
+	return newEvent(origin, eventType, agentRealmName, agentUserID, agentUsername, targetRealmName, &targetUserID, &targetUserName, details)
+}
+
+func NewEventOnUserFromContext(ctx context.Context, logger log.Logger, origin string, eventType string, targetRealmName string, targetUserID string, targetUserName string, details map[string]string) Event {
```
Comment (fperot74) [inline events/events.go:71]: NewEventOnAgentFromContext? how can the user be sure that the user in the context is considered as the user agent?

Comment (basbeu) [reply events/events.go:71]: The agent will be extracted from the context.
The target user will be taken from the params. 

NewEventOnAgentFromContext is not clearer to me and is not very meaningful

**Pull request merged**